### PR TITLE
Added MemoryStreamExtensions.cs

### PR DESCRIPTION
**Pull request by AraHaan**
This will help new people to clear their MemoryStreams a lot easier (possibly before disposing so it is not hard on GC with large MemoryStreams that might be on GC 2 or even the LOH (large object heap)).

### CODE REVIEW

**Comment by JustNrik** [comment]
Don't set the capacity to 0 just like that, if the user wants to keep using the stream after clearing it, it will allocate a new array that will be resized on demand, and if the array was already big enough to be in the LOH and the user keeps using it, then it will hurt performance a lot as it will be resized many times. Keep in mind that Clearing the stream doesn't meant that he won't use the stream anymore.

Instead, make some changes to allow the user to be more explicit with their intentions, like adding a parameter to set the new capacity, so you set it to 0 for the clean up and then you set it to the new capacity instead of leaving it at zero.

**Review by JustNrik — CHANGES_REQUESTED** [review]
Don't set the capacity to 0 just like that, if the user wants to keep using the stream after clearing it, it will allocate a new array that will be resized on demand, and if the array was already big enough to be in the LOH and the user keeps using it, then it will hurt performance a lot as it will be resized many times. Keep in mind that Clearing the stream doesn't meant that he won't use the stream anymore.

Instead, make some changes to allow the user to be more explicit with their intentions, like adding a parameter to set the new capacity, so you set it to 0 for the clean up and then you set it to the new capacity instead of leaving it at zero.

File: NetUtilities/System/IO/MemoryStreamExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        public static void Clear(this MemoryStream ms, int capacity)
+        {
+            if (ms == null)
+            {
+                throw new ArgumentNullException(nameof(ms));
+            }
+
+            // false if they are not equal or if capacity is 0.
+            bool changeCapacity = ms.GetBuffer().Length != capacity || capacity == 0;
+            Array.Clear(ms.GetBuffer(), 0, ms.GetBuffer().Length);
```
Comment (AraHaan) [inline NetUtilities/System/IO/MemoryStreamExtensions.cs:25]: I changed this as not sure if Array.Clear would only change only the local byte array, or the array owned by the MemoryStream when I did it the other way 😅. This way makes it more explicit but I feel I could save the length to a local variable too,

**Comment by AraHaan** [comment]
Should I also wrap ``SetLength`` too?

**Review by JustNrik — APPROVED** [review]

**Pull request merged**